sfcsample/VDecAccelVA.cpp

```diff
@@ -0,0 +1,811 @@
+ * * the following conditions:
+ * *
+ * * The above copyright notice and this permission notice (including the
+ * * next paragraph) shall be included in all copies or substantial portions
+ * * of the Software.
+ * *
+ * * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT.
+ * * IN NO EVENT SHALL PRECISION INSIGHT AND/OR ITS SUPPLIERS BE LIABLE FOR
+ * * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * */
+
+
+/**
+ * @file VDecAccelVA.cpp
+ * @brief LibVA decode accelerator implementation.
+ */
+
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/ioctl.h>
+#include <linux/fb.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <iostream>
+#include <assert.h>
+#include <algorithm>
+#include <thread>
+#include <chrono>
+#include <string.h>
+#include "VDecAccelVA.h"
+#include <va/va.h>
+#include <va/va_drm.h>
+#include <va/va_x11.h>
+
+#define VASUCCEEDED(err)    (err == VA_STATUS_SUCCESS)
+#define VAFAILED(err)       (err != VA_STATUS_SUCCESS)
+
+#define VA_SURF_ERR_MB_RANGE  1
+#define VA_SURF_ERR_MB_NUMBER 2
+#define VA_SURF_ERR_INVALID   -1
+
+#define VACONTEXTID_CLEAR         0x0FFFFFFF
+#define VACONTEXTID_DECODER       0x10000000
+#define VACONTEXTID_ENCODER       0x20000000
+#define VACONTEXTID_HUC           0x30000000
+
+mvaccel::VDecAccelVAImpl::VDecAccelVAImpl(void* device)
+    : m_vaDisplay(0)
+    , m_vaProfile(VAProfileNone)
+    , m_vaEntrypoint(VAEntrypointVLD)
+    , m_vaConfigID(0)
+    , m_vaContextID(0)
+    , m_surfaceType(VA_RT_FORMAT_YUV420)
+{
+    if (device)
+        m_vaDisplay = *(reinterpret_cast<VADisplay*>(device));
+
+    if (!m_vaDisplay)
+    {
+        printf("Invalid VADisplay\n");
+        delete this;
+        return;
+    }
+}
+
+mvaccel::VDecAccelVAImpl::VDecAccelVAImpl()
+    : m_vaDisplay(0)
+    , m_vaProfile(VAProfileNone)
+    , m_vaEntrypoint(VAEntrypointVLD)
+    , m_vaConfigID(0)
+    , m_vaContextID(0)
+    , m_surfaceType(VA_RT_FORMAT_YUV420)
+{
+}
+
+mvaccel::VDecAccelVAImpl::~VDecAccelVAImpl()
+{
+}
+
+//////////////////////////////////////////////////////////////////////////
+// VDecAccel interface
+//////////////////////////////////////////////////////////////////////////
+
+int mvaccel::VDecAccelVAImpl::Open()
+{
+    VAStatus vaStatus = VA_STATUS_SUCCESS;
+
+    //get display device
+    int MajorVer, MinorVer;
+    Display*    g_pDisplay = 0;
+    g_pDisplay = XOpenDisplay(":0.0");
+
+    int drm_fd = -1;
+
+    if (g_pDisplay != NULL) {
+        m_vaDisplay = vaGetDisplay(g_pDisplay);
+        vaInitialize(m_vaDisplay, &MajorVer, &MinorVer);
+    }
+
+    if (vaStatus != VA_STATUS_SUCCESS) {
```

Comment:
vaStatus here is alwasy VA_STATUS_SUCCESS

Reply:
initialize the vaStatus as VA_STATUS_ERROR_UNKNOWN in line 100